website/src/main/java/com/google/step/servlets/DataServlet.java

```diff
@@ -37,17 +41,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    DatastoreServiceFactory.getDatastoreService().put(getOrgEntityFrom(request));
-    response.sendRedirect("/index.html");
+    OrganizationInfo submission = new OrganizationInfo(request);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.prepare(submission.getQueryForDuplicates()).asList(FetchOptions.Builder.withDefaults()).forEach((duplicate -> {
+      submission.compare(duplicate);
+      datastore.delete(duplicate.getKey());
+    }));
+    if (submission.isValid()) {
+      datastore.put(submission.getEntity());
+      response.sendRedirect("/index.html");
+    } else {
+      response.sendRedirect("/");
+    }
```

Comment:
It may make more sense for a duplicate entry to be a failure case for now, and to add an "Update Info" flow when a duplicate is detected (as a new feature in a later PR).

Merging can work but I think it might be better to provide more fine-grained control.